tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,285 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @file tizen-api.h
+ * @date 07 March 2019
+ * @brief Tizen NNStreamer/Pipeline(main) C-API Header.
+ *        This allows to construct and control NNStreamer pipelines.
+ * @see	https://github.com/nnsuite/nnstreamer
+ * @author MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#ifndef __TIZEN_NNSTREAMER_API_MAIN_H__
+#define __TIZEN_NNSTREAMER_API_MAIN_H__
+
+#include <nnstreamer/tensor_typedef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+/**
+ * @addtogroup CAPI_ML_NNSTREAMER_PIPELINE_MODULE
+ * @{
+ */
+
+/**
+ * @brief A handle of an NNStreamer pipeline.
+ * @since_tizen 5.5
+ */
+typedef struct nns_pipeline *nns_pipeline_h;
+
+/**
+ * @brief A handle of a "sink node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_sink *nns_sink_h;
+
+/**
+ * @brief A handle of a "src node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_src *nns_src_h;
+
+/**
+ * @brief A handle of a "switch" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_switch *nns_switch_h;
+
+/**
+ * @brief A handle of a "valve node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_valve *nns_valve_h;
+
+/**
+ * @brief Enumeration for the error codes of NNStreamer Pipelines.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_ERROR_NONE				= TIZEN_ERROR_NONE, /**< Success! */
+  NNS_ERROR_INVALID_PARAMETER			= TIZEN_ERROR_INVALID_PARAMETER, /**< Invalid parameter */
+  NNS_ERROR_NOT_SUPPORTED			= TIZEN_ERROR_NOT_SUPPORTED, /**< The feature is not supported */
+} nns_error_e;
+
+/**
+ * @brief Enumeration for buffer deallocation policies.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_BUF_FREE_BY_NNSTREAMER			= 0, /**< Default. Application should not deallocate this buffer. NNStreamer will deallocate when the buffer is no more needed */
+  NNS_BUF_NO_NOT_FREE				= 1, /**< This buffer is not to be freed. */
+} nns_buf_policy;
+
+/**
+ * @brief Enumeration for nns pipeline status
+ * @detail Refer to https://gstreamer.freedesktop.org/documentation/design/states.html
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI) Fill with GST's status values.
+ * @todo Check the consistency against MMFW APIs.
+ */
+typedef enum {
+  NNS_PIPELINE_UNKNOWN				= 0, /**< Unknown status. Maybe not constructed? */
+  NNS_PIPELINE_NULL,				= 1, /**< GST-State "Null" */
+  NNS_PIPELINE_READY,				= 2, /**< GST-State "Ready" */
+  NNS_PIPELINE_PAUSED,				= 3, /**< GST-State "Paused" */
+  NNS_PIPELINE_PLAYING,				= 4, /**< GST-State "Playing" */
+} nns_pipeline_status;
```

Comment:
In order to avoid confusing, how about using "nns_pipeline_state", which is a same term used in gstreamer, instead of "nns_pipeline_status".

FYI, [states](https://gstreamer.freedesktop.org/documentation/design/states.html) and [status](https://gstreamer.freedesktop.org/documentation/design/stream-status.html) in gstreamer.

Reply:
Updated